subsys/bluetooth/mesh/shell/shell_sensor_cli.c

```diff
@@ -0,0 +1,489 @@
+
+	if (sensor_type == NULL) {
+		return -ENOENT;
+	}
+
+	int err = bt_mesh_sensor_cli_desc_get(cli, NULL, sensor_type, &rsp);
+
+	descriptor_print(shell, err, &rsp);
+	return err;
+}
+
+static void cadence_print(const struct shell *shell, int err,
+			  struct bt_mesh_sensor_cadence_status *rsp)
+{
+	if (!err) {
+		shell_print(shell, "fast period div: %d\nmin interval: %d", rsp->fast_period_div,
+			    rsp->min_int);
+		shell_fprintf(shell, SHELL_NORMAL, "delta threshold: { type: %d, up: ",
+			      rsp->threshold.delta.type);
+		shell_model_print_sensorval(shell, &rsp->threshold.delta.up);
+		shell_fprintf(shell, SHELL_NORMAL, ", down: ");
+		shell_model_print_sensorval(shell, &rsp->threshold.delta.down);
+		shell_fprintf(shell, SHELL_NORMAL,
+			      " }\nfast cadence range: { cadence inside: %d, lower boundary: ",
+			      rsp->threshold.range.cadence);
+		shell_model_print_sensorval(shell, &rsp->threshold.range.low);
+		shell_fprintf(shell, SHELL_NORMAL, ", upper boundary: ");
+		shell_model_print_sensorval(shell, &rsp->threshold.range.high);
+		shell_print(shell, " }");
+	}
+}
+
+static int cmd_cadence_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_SENSOR_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_sensor_cli *cli = mod->user_data;
+
+	struct bt_mesh_sensor_cadence_status rsp;
+	uint32_t sensor_id = (uint16_t)strtol(argv[1], NULL, 0);
+	const struct bt_mesh_sensor_type *sensor_type = bt_mesh_sensor_type_get(sensor_id);
+
+	if (sensor_type == NULL) {
+		return -ENOENT;
+	}
+
+	int err = bt_mesh_sensor_cli_cadence_get(cli, NULL, sensor_type, &rsp);
+
+	cadence_print(shell, err, &rsp);
+	return err;
+}
+
+static int cadence_set(const struct shell *shell, size_t argc, char *argv[], bool acked)
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_SENSOR_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_sensor_cli *cli = mod->user_data;
+
+	uint32_t sensor_id = (uint16_t)strtol(argv[1], NULL, 0);
+	const struct bt_mesh_sensor_type *sensor_type = bt_mesh_sensor_type_get(sensor_id);
+
+	if (sensor_type == NULL) {
+		return -ENOENT;
+	}
+
+	struct bt_mesh_sensor_cadence_status cadence = {
+		.fast_period_div = (uint8_t)strtol(argv[2], NULL, 0),
+		.min_int = (uint8_t)strtol(argv[3], NULL, 0),
+		.threshold = {
+			.delta = { .type = (int)strtol(argv[4], NULL, 0) },
+			.range = { .cadence = (int)strtol(argv[7], NULL, 0) }
+		}
+	};
+	struct bt_mesh_sensor_cadence_status rsp;
+
+	int err = shell_model_str2sensorval(argv[5], &cadence.threshold.delta.up);
+
+	if (err) {
+		return err;
+	}
+	err = shell_model_str2sensorval(argv[6], &cadence.threshold.delta.down);
+	if (err) {
+		return err;
+	}
+	err = shell_model_str2sensorval(argv[8], &cadence.threshold.range.low);
+	if (err) {
+		return err;
+	}
+	err = shell_model_str2sensorval(argv[9], &cadence.threshold.range.high);
+	if (err) {
+		return err;
+	}
```

Comment:
More of a question than a comment really. I think this looks fine, but eventually we want to check every string conversion to avoid triggering a mesh message on invalid input. To avoid very long stretches of error checks would it be possible to do something like this? In this particular case we don't care what error occured, only that at least one did occur.
```suggestion
        int err = 0;
        
	err |= shell_model_str2sensorval(argv[5], &cadence.threshold.delta.up);
	err |= shell_model_str2sensorval(argv[6], &cadence.threshold.delta.down);
	err |= shell_model_str2sensorval(argv[8], &cadence.threshold.range.low);
	err |= shell_model_str2sensorval(argv[9], &cadence.threshold.range.high);
	
	if (err) {
		return -EINVAL;
	}
```

Reply:
Yeah, that would be possible. I think I'll keep this like it is for now, but we can keep this in mind for when we swap the parsing functions later